apps/stats/tests/test_views.py

```diff
@@ -878,3 +882,104 @@ def test_report_view_xss(self):
 
         req = RequestFactory().get('/', last='<alert>')
         assert views.get_report_view(req) == {}
+
+
+class ArchiveTestCase(APIAuthTestCase):
+    fixtures = ['base/addon_3615']
+
+    def setUp(self):
+        self.user = UserProfile.objects.get(email='del@icio.us')
+        self.api_key = self.create_api_key(self.user, str(self.user.pk) + ':f')
+        self.addon = Addon.objects.get(pk=3615)
+        self.theme_update_count = ThemeUpdateCount(
+            addon_id=3615, date='2016-01-18', count=123)
+
+    def tearDown(self):
+        self.clean_up_files()
+
+    def clean_up_files(self):
+        path = os.path.join(views.storage.location, '3615')
+        if os.path.isdir(path):
+            shutil.rmtree(path)
+
+    def get(self, url=None):
+        return self.client.get(url, HTTP_AUTHORIZATION=self.authorization())
+
+    def test_list_not_authenticated(self):
+        # Use self.client.put so that we don't add the authorization header.
+        response = self.client.get(
+            reverse('stats.archive_list', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+
+        assert response.status_code == 401
+
+    def test_list_does_not_own_addon(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        response = self.get(
+            reverse('stats.archive_list', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+        assert response.status_code == 403
+
+    def test_list(self):
+        self.user = UserProfile.objects.create(
```

Comment:
`self.user` is being used by `APIAuthTestCase` and used as a default for non-authenticated tests. I'm creating the new user again every time to explicitly make sure I have an authorized user.